plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDataSource.java

```diff
@@ -792,7 +795,50 @@ public boolean isMariaDB() {
         return MySQLConstants.DRIVER_CLASS_MARIA_DB.equals(
             getContainer().getDriver().getDriverClassName());
     }
-
+    
+    public boolean isTiDB() {
+        return MySQLConstants.DRIVER_ID_TIDB.equals(
+            getContainer().getDriver().getId());
+    }
+    
+    // TiDB version string looks like: `5.7.25-TiDB-v6.1.0`
+    // JDBC will return major version 5 and minor version 7
+    // But the real TiDB version is v6.1.0
+    public boolean isTiDBServerVersionAtLeast (int major, int minor) {
+    	if (!isTiDB()) {
+    		return false;
+    	}
+    	
+    	String tidbVersion = this.getInfo().getDatabaseProductVersion();
+    	String[] tidbVersionArray = tidbVersion.split("-");
+    	if (tidbVersionArray.length != 3 || !tidbVersionArray[1].equals("TiDB")) {
+    		// It means not a TiDB server actually
+    		return false;
+    	}
+    	
+    	// drop first char "v"
+    	String realTiDBVersion = tidbVersionArray[2].substring(1);
+    	String[] realTiDBVersionArray = realTiDBVersion.split("\\.");
+    	if (realTiDBVersionArray.length != 3) {
+    		// It means not a TiDB server actually
+    		return false;
+    	}
+    	
+    	try {
+    		int serverMajor = Integer.parseInt(realTiDBVersionArray[0]);
+    		int serverMinor = Integer.parseInt(realTiDBVersionArray[1]);
+    		
+    		// target major version is less than server 
+    		// OR
+    		// major version equals AND target minor version is equal or less than server
+    		return (major < serverMajor || (major == serverMajor && minor <= serverMinor));
+    	} catch (NumberFormatException e){
+    		// It means not a TiDB server actually
+    		log.error("TiDB server version not a number, error: " + e.getMessage());
+    		return false;
+    	}
```

Comment:
All this block should be 4 spaces indented.

Reply:
OK, I'll replace all tabs to 4 spaces.